packages/react-native-bridge/index.js

```diff
@@ -396,4 +396,32 @@ export function requestPreview() {
 	RNReactNativeGutenbergBridge.requestPreview();
 }
 
+/**
+ * Request the host app provide the latest block type impression counts.
+ *
+ * @param {Function} callback      Callback invoked with object containing counts for each block type.
+ * @param {Array}    newBlockTypes The block types to consider new.
+ * @return {void}
+ */
+export function requestBlockTypeImpressions( callback, newBlockTypes = [] ) {
+	return RNReactNativeGutenbergBridge.requestBlockTypeImpressions(
+		newBlockTypes,
+		callback
+	);
+}
+
+/**
+ * Request the host app set updated impression count for a given block type identified by name.
+ *
+ * @param {string} name  The block type name to update.
+ * @param {number} count The number of impressions for the block type.
```

Comment:
Of course, we'd define a threshold score and when a block's score surpasses this, the badge would no longer be shown.

Reply:
> I'm not sure where to leave this comment, but I wanted to raise how the meaning given to the word "impression" in this project doesn't feel intuitive to me. 

> The closest definition of the word "impression" that seems to apply to the context here is the marketing term which I understand as "an impression occurs when something has been eyeballed by the user". 

@Guarani thank you for raising this. I agree the use of the term "impression" is leaning on a loose definition. In reality, it currently represents remaining "actions" before a block is stale. E.g. "inserting" immediately marks a block type stale, #33528 plans to incrementally mark a block type stale when a post is published. As you mentioned, we could also incrementally remove newness when a block type is viewed within the inserter or editor. 

I am definitely open to renaming it to improve clarity. I have found identifying a clear name to be challenging given the various actions that modify newness. Would `noveltyScore` or `newnessRating` better communicate the intent? 

> [...] Given this point of view, I expected the impression count to _increase_ each time a block presented to the user (be it in the block inserter or the editor itself). It does the opposite though, which leads to the confusion I had.
>
> What I see here looks more like a "novelty factor", although that's doesn't describe it well either.
> Have we considered flipping the way it's used by giving each of the [new blocks](https://github.com/WordPress/gutenberg/pull/33119/files?file-filters%5B%5D=.java&file-filters%5B%5D=.js&file-filters%5B%5D=.m&file-filters%5B%5D=.swift#diff-3286cc974baacc1ef30a6c7d6c18ae3ea400c86e1223c88029c1ea15051d65ecR290) a "usage score"? For example,
> 
> * a view in the block inserter (i.e. the block is displayed to the user): plus 2 points
> * an insert (i.e. the block is inserted by the user) via the block inserter or slash inserter: plus 5 points
> * a view in a post (i.e. the user opened a post in the editor and it contained the block): plus 1 points

In regards to the current inverted nature of the count, that was done for simplicity. Specifically, I think it is expected/easy to communicate the counter will stop decrementing when it reaches zero. If we count up, we'd need to define an arbitrary maximum for the count. If counting down is truly confusing, we could flip it and define a maximum. 

One alternative could be renaming to communicate that the value counts downward. That could be one of the aforementioned names to avoid the "impressions" term (e.g. `noveltyScore`) or a prefix to communicate the value will be reduced (e.g. `remainingImpressions`, `availableImpressions`, `unusedImpressions`). WDYT? 

> Only new blocks would have a score (blocks not in `NEW_BLOCK_TYPES` wouldn't have a score, which is how the current implementation works I believe).

From my perspective this relates to the simplicity of the decrementing down to zero. Checking if a given block type count is less than zero is more straightforward than checking if a block type has a score (1) set _and_  (2) less than `<arbitary-maximum>`. That is likely a personal opinion, and not one that I hold onto strongly. WDYT? 

I'd also be curious to see @jhnstn's thoughts, as he was the original source of the impression count idea.

Reply:
>  I expected the impression count to increase each time a block presented to the user (be it in the block inserter or the editor itself). It does the opposite though, which leads to the confusion I had.

Thanks for mentioning this @guarani . I had the same confusion, meant to bring it up, and promptly forgot to do that. 😞 

>  I have found identifying a clear name to be challenging given the various actions that modify newness. 

<nod><nod> I suspect that my inability to think of a good name is one of the reasons I conveniently "forgot" to bring this up. 😆 

> In regards to the current inverted nature of the count, that was done for simplicity. Specifically, I think it is expected/easy to communicate the counter will stop decrementing when it reaches zero. If we count up, we'd need to define an arbitrary maximum for the count. If counting down is truly confusing, we could flip it and define a maximum.

I also think that decrementing the count makes sense. 

> One alternative could be renaming to communicate that the value counts downward. That could be one of the aforementioned names to avoid the "impressions" term (e.g. noveltyScore) or a prefix to communicate the value will be reduced (e.g. remainingImpressions, availableImpressions, unusedImpressions). WDYT?

The prefix names all seem like nice improvements. We could even expand them to be even more descriptive with something like `remainingNewBlockImpressions`. I do not have any strong feelings on the specific name though.

Reply:
I would stick with impressions as the term but I can see where there might be some confusion. Specifically I see how decrementing an `impression` value seems odd if encountering a snapshot of the system i.e. I see the state of an app install and a block has an impression count of `N`. Without knowing the semantics of "impression" in this context I could conclude that the block has made `N` impressions. I still wouldn't know what the impression threshold is ( or if one exists at all) but assuming impression count going up is plausible.  

Personally I don't see it this way. I see it as the block _having_ a set amount of impressions to give. Therefore I would assume that count would be going down. I am biased since I already envisioned this system based on prior experience where a "newness" value goes down.

That said the term is too subjective to know by looking purely at the data store to know which way the system moves that value. In either case it would behoove us to reify that in the name such as : `impressionsMade` or `impressionsRemaining`. Right away I don't have to spell out which term should be used in an incrementing or decrementing system.  I have no preference on the actual name, prefix or suffix, or what not. 

Naming and caching, all the tough stuff.

I would also stick with a decrementing system since it's simpler and less prone to accidentally making something old looking new again.

Reply:
Thanks everyone for sharing their thoughts. I agree that there's no clear way to improve this. I don't feel strongly that this _needs_ improving either, so I'm happy to leave it as-is.
I think my concern about it being unintuitive still stands in my mind, but it's hard to name things in a way that lends itself to the correct interpretation to everyone.
While renaming to `remainingImpressions` is probably an option in my mind, but not sure it's worth it tbh. So I'm happy to leave this as-is for now.

Reply:
I'm happy to mark this thread resolved, but want to give @mchowning a chance to reply first since he also shared thoughts here.

Reply:
I would rename the variable to something like `remainingImpressions`, but I do not view it as a blocker. I'm fine resolving this thread if you'd rather leave it as-is.

Reply:
If I am honest, I understand the confusion with the term “impression,” but I do not believe we have an alternative name that meaningfully reduces confusion, at least not enough to rename all the values across three repos at this time. 

I spoke with @mchowning via chat regarding this just now. Based on that chat and the most recent comments from other team members in this thread, I am marking this thread resolved and will leave the implementation as-is. Please feel free to reopen this discussion if any of you feel this is a mistake. 

Thank you. 🙇🏻